privaterelay/tests/utils_tests.py

```diff
@@ -16,60 +15,305 @@
 
 from ..plans import get_premium_country_language_mapping
 from ..utils import (
+    AcceptLanguageError,
     flag_is_active_in_task,
-    get_premium_country_lang,
+    get_countries_info_from_request_and_mapping,
+    guess_country_from_accept_lang,
 )
 
 
-class GetPremiumCountryLangTest(TestCase):
-    def setUp(self):
-        self.mapping = get_premium_country_language_mapping(None)
-
-    def test_get_premium_country_lang(self):
-        cc, lang = get_premium_country_lang("en-au,", self.mapping)
-        assert cc == "au"
-        assert lang == "en"
-
-        cc, lang = get_premium_country_lang("en-us,", self.mapping)
-        assert cc == "us"
-        assert lang == "en"
-
-        cc, lang = get_premium_country_lang("de-be,", self.mapping)
-        assert cc == "be"
-        assert lang == "de"
-
-        cc, lang = get_premium_country_lang("de-be,", self.mapping, "at")
-        assert cc == "at"
-        assert lang == "de"
-
-    def test_en_fallback(self) -> None:
-        cc, lang = get_premium_country_lang("en,", self.mapping)
-        assert cc == "us"
-        assert lang == "en"
-
-    def test_first_lang_fallback_two_parts(self) -> None:
-        accept_lang = "sgn-us,"  # American Sign Language
-        cc, lang = get_premium_country_lang(accept_lang, self.mapping)
-        assert cc == "us"
-        assert lang == "en"
-
-    def test_first_lang_fallback_three_parts(self) -> None:
-        accept_lang = "sgn-ch-de,"  # Swiss German Sign Language
-        cc, lang = get_premium_country_lang(accept_lang, self.mapping)
-        assert cc == "ch"
-        assert lang == "fr"
-
-    def test_eu_country_expansion_active(self) -> None:
-        mapping = get_premium_country_language_mapping(eu_country_expansion=True)
-        cc, lang = get_premium_country_lang("et-ee", mapping)
-        assert cc == "ee"
-        assert lang == "et"
-
-    def test_eu_country_expansion_inactive(self) -> None:
-        mapping = get_premium_country_language_mapping(eu_country_expansion=False)
-        cc, lang = get_premium_country_lang("et-ee", mapping)
-        assert cc == "ee"
-        assert lang == "en"
+@pytest.mark.parametrize(
```

Comment:
As soon as I read the PR description, I knew there was about to be an *EPIC* `parametrize` decorator in here. 🤯